README.md

```diff
@@ -5,102 +5,165 @@
+export ZONE=us-east1-b
+export AUTH_FILE=<path to service-account.json>
+
+./prombench gke cluster create -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID \
+    -v ZONE:$ZONE -v CLUSTER_NAME:$CLUSTER_NAME -f components/prow/cluster.yaml
+```
+
+### Deploy Prometheus-Meta & Grafana
+---
+
+This is used for collecting and displaying the test results.
+
+- Set the following environment variables
+```
+export GCLOUD_SERVICEACCOUNT_CLIENTID=<email-id present in service-account.json>
+export GRAFANA_ADMIN_PASSWORD=password
+```
+**Note:** The `GCLOUD_SERVICEACCOUNT_CLIENTID` is used to grant cluster-admin-rights to the service-account. This is needed to create RBAC roles on GKE.
+
+- Deploy the [nginx-ingress-controller](https://github.com/kubernetes/ingress-nginx) which will be used to access Prometheus-Meta & Grafana.
+```
+./prombench gke resource apply -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID -v ZONE:$ZONE \
+    -v CLUSTER_NAME:$CLUSTER_NAME -v GCLOUD_SERVICEACCOUNT_CLIENTID:$GCLOUD_SERVICEACCOUNT_CLIENTID \
+    -f components/prow/manifests/rbac.yaml -f components/prow/manifests/nginx-controller.yaml
+```
+
+- Export the nginx-ingress-controller IP address.
+```
+kubectl get ingress ing -o go-template='{{ range .status.loadBalancer.ingress}}{{.ip}}{{ end }}'
+
+export INGRESS_IP=$(kubectl get ingress ing -o go-template='{{ range \
+    .status.loadBalancer.ingress}}{{.ip}}{{ end }}')
+```
+
+- Deploy Prometheus-meta & Grafana.
+```
+./prombench gke resource apply -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID \
+    -v ZONE:$ZONE -v CLUSTER_NAME:$CLUSTER_NAME -v INGRESS_IP:$INGRESS_IP \
+    -v GRAFANA_ADMIN_PASSWORD:$GRAFANA_ADMIN_PASSWORD -f components/prombench/manifests/results
+```
+
+- The services will be accessible at:
+  * Grafana :: http://<INGRESS_IP>/grafana
+  * Prometheus :: http://<INGRESS_IP>/prometheus-meta
+
+### Start a test
+---
+
+- Set the following environment variables.
+```
+export RELEASE=<master or any prometheus release(ex: v2.3.0) >
+export PR_NUMBER=<PR number to benchmark the release with>
+```
+
+- Create the nodepools for the k8s objects
+```
+./prombench gke nodepool create -a ${AUTH_FILE} \
+    -v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} \
+    -v PR_NUMBER:${PR_NUMBER} -f components/prombench/nodepools.yaml
+```
+
+- Deploy the k8s objects
+```
+./prombench gke resource apply -a ${AUTH_FILE} \
+    -v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} \
+    -v PR_NUMBER:${PR_NUMBER} -v RELEASE:${RELEASE} \
+    -f components/prombench/manifests/benchmark
+```
+
 ## How to trigger tests on Github
-A Prometheus maintainer can comment as follows to benchmark a PR:
-- `/benchmark` equivalent to `/benchmark master`  - benchmark PR with the master branch.
-- `/benchmark 2.4.0` - benchmark the PR with the 2.4.0 release(could be any release). Don't prepend `v` to the release version.
 
-To cancel benchmarking, a mantainer should comment `/benchmark cancel`.
+### Prerequisites
```

Comment:
we can simplify this as:

### Create a k8s cluster
Follow the steps from the [manual setup](link).

 
### Prerequisites -change to-> Setup the github API